Automattic/wp-calypso
# Move posts-related selectors under state/posts

This PR moves all posts-related selectors under `state/posts/selectors`, since posts state has already been modularised.

For consistency, it also turns default exports into named exports.

All existing code is updated to use the new paths.

#### Changes proposed in this Pull Request

* Move all posts-related selectors under `state/posts/selectors`
* Rewrite selectors to use named exports rather than default exports
* Update all importers accordingly
* Minor linting-related changes

#### Testing instructions

There is no functionality change in this PR, so it should be enough to ensure that all unit tests pass, and that there are no build warnings related to missing exports or the like.
